### ISSUE TITLE
Vercel Sharp

### ISSUE THREAD

**Issue by ohrrkan** [withastro/astro#9592]
### Astro Info

```block
KO : "astro": "4.0.8", "astro": "4.0.9". 
Ok : "astro": "4.0.7",


both with "@astrojs/vercel": "6.1.2",
```

### Describe the Bug

built-in `<Image />` are no more correctly generated by sharp when build with Vercel integration. 

installing sharp module do not solve it. 

astro build give this new compilation warning  : 

[@astrojs/vercel] The module "@img/sharp-wasm32/versions" inside the file "/node_modules/sharp/lib/utility.js" couldn't be resolved. This may not be a problem, but it's worth checking.
[@astrojs/vercel] The module "@img/sharp-libvips-/package" inside the file "/node_modules/sharp/lib/sharp.js" couldn't be resolved. This may not be a problem, but it's worth checking.
[@astrojs/vercel] The module "@img/sharp-libvips-dev/include" inside the file "/node_modules/sharp/lib/libvips.js" couldn't be resolved. This may not be a problem, but it's worth checking.
[@astrojs/vercel] The module "@img/sharp-libvips-dev/cplusplus" inside the file "/node_modules/sharp/lib/libvips.js" couldn't be resolved. This may not be a problem, but it's worth checking.

### What's the expected result?

Why `<Image />` do not work anymore with Vercel 

### Link to Minimal Reproducible Example

[vercel](https://github.com/ohrrkan/test-astro)

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nickgraffis** [comment]
Hey,

Adding a reproduction for what I think is the issue @ohrrkan is describing.

https://github.com/nickgraffis/astro-vercel-bug

Steps that I took:

1. `npm create astro@latest`
2. `npx astro add vercel`
3. `npm run build`


Resulting error:

```
13:40:42 [build] Building server entrypoints...
13:40:42 [ERROR] [vite] [vite]: Rollup failed to resolve import "sharp" from "/Users/nickgraffis/Nick/visible-virgo/node_modules/astro/dist/assets/services/sharp.js".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
[vite]: Rollup failed to resolve import "sharp" from "/Users/nickgraffis/Nick/visible-virgo/node_modules/astro/dist/assets/services/sharp.js".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
  Stack trace:
    at viteWarn (file:///Users/nickgraffis/Nick/visible-virgo/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:66953:27)
    at onwarn (file:///Users/nickgraffis/Nick/visible-virgo/node_modules/vite/dist/node/chunks/dep-R0I0XnyH.js:66698:13)
    at Object.logger [as onLog] (file:///Users/nickgraffis/Nick/visible-virgo/node_modules/rollup/dist/es/shared/node-entry.js:19028:9)
    at ModuleLoader.resolveDynamicImport (file:///Users/nickgraffis/Nick/visible-virgo/node_modules/rollup/dist/es/shared/node-entry.js:18032:58)
```

**Comment by nickgraffis** [comment]
For what it's worth a possible fix is here https://github.com/nickgraffis/astro-vercel-bug/tree/a-possible-fix

Added this plugin for rollup

```
import { defineConfig } from "astro/config";
import { nodeExternals } from "rollup-plugin-node-externals";
import vercel from "@astrojs/vercel/serverless";

// https://astro.build/config
export default defineConfig({
  output: "server",
  adapter: vercel(),
  build: {
    rollupOptions: {
      plugins: [nodeExternals()],
    },
  },
});
```

then had to move the following to `devDependencies` in `package.json`

```
    "@astrojs/check": "^0.3.4",
    "@astrojs/vercel": "^6.1.2",
    "astro": "^4.1.0",
    "typescript": "^5.3.3"
```

**Comment by domingouribe** [comment]
Hello.  I had the same problem.  I solved it by first updating Node to version 20.10.  I was working with version 18. Then I updated Astro and the dependencies.  And finally I manually installed Sharp with npm (Not with pnpm), and it worked.

Here is a simple project with the solution applied: https://github.com/domingouribe/news-card.git

**Comment by ohrrkan** [comment]
Thanks @domingouribe but it do not solved the issue with Vercel. No issue with a node build but with Vercel integration, sharp will not work. 

here is the example base on your example (https://github.com/ohrrkan/test-astro) with the add of Vercel. You can see the difference between "npm run build" : KO (this is with Vercel adapter)

"npm run build:node" : OK (with node adapter)


Upload on Vercel server this will lead to this kind of error but as you can see sharp module is install. Getting back to astro 4.0.7 solved the issue. so it is probably a change made in the 4.0.8.

```
Could not process image request: MissingSharp: Could not find Sharp. Please install Sharp (`sharp`) manually into your project or migrate to another image service.
    at loadSharp (file:///var/task/vercel/path0/dist/chunks/astro/assets-service_zQugzRVo.mjs:327:11)
```

**Comment by ohrrkan** [comment]
ok I find one of the reason. I was not using    `imageService: true,` in the config. This do not explain the warning when building and why it was working before 4.07 but it work with this following astro.config.mjs

```
import { defineConfig } from 'astro/config';
import vercel from '@astrojs/vercel/static';

export default defineConfig({
  // ...
  output: 'static',
  adapter: vercel({
    imageService: true,
  }),
});
```


as the warning end with : "This may not be a problem, but it's worth checking." I will close the issue

**Comment by ohrrkan** [comment]
at the end it is a duplicate of #9345 

and my solving :
 ```
 (vercel({
    imageService: true,
  }),) 
```
is more a bypass than a real solution of the original issue => it use the Vercel service in place of the sharp astro one. for now I solved it by getting back to sharp "^0.32"
